api/src/test/java/org/openmrs/module/fhir2/api/dao/impl/FhirEncounterDaoImplTest.java

```diff
@@ -72,21 +100,183 @@ public void shouldReturnNullWithUnknownEncounterUuid() {
+		
+		assertThat(results, Matchers.notNullValue());
+		assertThat(results, is(empty()));
+	}
+	
+	@Test
+	public void searchForEncounters_shouldSearchForEncountersByParticipantIdentifier() {
+		ReferenceParam participantReference = new ReferenceParam();
+		
+		participantReference.setChain(Practitioner.SP_IDENTIFIER);
+		participantReference.setValue(PATIENT_IDENTIFIER);
+		
+		Collection<Encounter> results = dao.searchForEncounters(null, null, participantReference, null);
+		
+		assertThat(results, Matchers.notNullValue());
+		assertThat(results, not(empty()));
+		assertThat(results.iterator().next().getEncounterProviders().size(), greaterThanOrEqualTo(1));
+		assertThat(results.iterator().next().getEncounterProviders().iterator().next().getProvider(), notNullValue());
+		assertThat(results.iterator().next().getEncounterProviders().iterator().next().getProvider().getIdentifier(),
+		    equalTo(PARTICIPANT_IDENTIFIER));
 	}
 	
+	@Test
+	public void searchForEncounters_shouldSearchForEncountersByParticipantGivenName() {
+		ReferenceParam participantReference = new ReferenceParam();
+		
+		participantReference.setValue(PARTICIPANT_GIVEN_NAME);
+		participantReference.setChain(Practitioner.SP_GIVEN);
+		
+		Collection<Encounter> results = dao.searchForEncounters(null, null, participantReference, null);
+		
+		assertThat(results, Matchers.notNullValue());
+		assertThat(results, not(empty()));
+		assertThat(results.iterator().next().getEncounterProviders().size(), greaterThanOrEqualTo(1));
+		assertThat(results.iterator().next().getEncounterProviders().iterator().next().getProvider(), notNullValue());
+		assertThat(results.iterator().next().getEncounterProviders().iterator().next().getProvider().getPerson()
+		        .getPersonName().getGivenName(),
+		    equalTo(PARTICIPANT_GIVEN_NAME));
+		
+	}
+	
+	@Test
+	public void searchForEncounters_shouldSearchForEncountersByParticipantFamilyName() {
+		ReferenceParam participantReference = new ReferenceParam();
+		
+		participantReference.setValue(PARTICIPANT_FAMILY_NAME);
+		participantReference.setChain(Practitioner.SP_FAMILY);
+		
+		Collection<Encounter> results = dao.searchForEncounters(null, null, participantReference, null);
+		
+		assertThat(results, Matchers.notNullValue());
+		assertThat(results, not(empty()));
+		assertThat(results.iterator().next().getEncounterProviders().size(), greaterThanOrEqualTo(1));
+		assertThat(results.iterator().next().getEncounterProviders().iterator().next().getProvider(), notNullValue());
+		assertThat(results.iterator().next().getEncounterProviders().iterator().next().getProvider().getPerson()
+		        .getPersonName().getFamilyName(),
+		    equalTo(PARTICIPANT_FAMILY_NAME));
+		
+	}
+	
+	@Test
+	public void searchForEncounters_shouldSearchForEncountersByParticipantName() {
+		ReferenceParam subjectReference = new ReferenceParam();
+		
+		subjectReference.setValue(PATIENT_FULL_NAME);
```

Comment:
This constant should be `PRACTITIONER_FULL_NAME` even if it has the same value as `PATIENT_FULL_NAME`